ConsumerApi.Tests.Integration/StepDefinitions/PnsRegistrationStepDefinitions.cs

```diff
@@ -1,62 +1,57 @@
-using System.Net;
-using Backbone.ConsumerApi.Tests.Integration.API;
+﻿using Backbone.BuildingBlocks.SDK.Endpoints.Common.Types;
+using Backbone.ConsumerApi.Sdk;
+using Backbone.ConsumerApi.Sdk.Authentication;
+using Backbone.ConsumerApi.Sdk.Endpoints.PushNotifications.Types.Requests;
+using Backbone.ConsumerApi.Sdk.Endpoints.PushNotifications.Types.Responses;
 using Backbone.ConsumerApi.Tests.Integration.Configuration;
-using Backbone.ConsumerApi.Tests.Integration.Helpers;
-using Backbone.ConsumerApi.Tests.Integration.Models;
-using Backbone.Crypto.Abstractions;
-using Backbone.Modules.Devices.Application.PushNotifications.Commands.UpdateDeviceRegistration;
+using Backbone.Tooling;
 using Microsoft.Extensions.Options;
-using Newtonsoft.Json;
-using RequestConfiguration = Backbone.ConsumerApi.Tests.Integration.Models.RequestConfiguration;
 
 namespace Backbone.ConsumerApi.Tests.Integration.StepDefinitions;
 
 [Binding]
 [Scope(Feature = "PUT /Devices/Self/PushNotifications")]
-internal class PnsRegistrationStepDefinitions : BaseStepDefinitions
+internal class PnsRegistrationStepDefinitions
 {
-    // keep in mind: these tests use DummyPushService so they will not execute the implemented code
+    private Client? _sdk;
+    private readonly ClientCredentials _clientCredentials;
+    private readonly HttpClient _httpClient;
+    private ApiResponse<UpdateDeviceRegistrationResponse>? _response;
 
-    private readonly PushNotificationsApi _pnsRegistrationsApi;
-    private HttpResponse<UpdateDeviceRegistrationResponse>? _response;
+    public PnsRegistrationStepDefinitions(HttpClientFactory factory, IOptions<HttpConfiguration> httpConfiguration)
+    {
+        _httpClient = factory.CreateClient();
+        _clientCredentials = new ClientCredentials(httpConfiguration.Value.ClientCredentials.ClientId, httpConfiguration.Value.ClientCredentials.ClientSecret);
+    }
 
-    public PnsRegistrationStepDefinitions(
-        IOptions<HttpConfiguration> httpConfiguration, ISignatureHelper signatureHelper, ChallengesApi challengesApi, IdentitiesApi identitiesApi, DevicesApi devicesApi, PushNotificationsApi pnsRegistrationsApi) :
-        base(httpConfiguration, signatureHelper, challengesApi, identitiesApi, devicesApi)
+    [Given("the user is authenticated")]
+    public async Task GivenTheUserIsAuthenticated()
     {
-        _pnsRegistrationsApi = pnsRegistrationsApi;
+        _sdk = await Client.CreateForNewIdentity(_httpClient, _clientCredentials, PasswordHelper.GeneratePassword(20, 26));
```

Comment:
We should use a hardcoded password instead. Same for any other occurrence that might exist